### DISCUSSION TITLE
ODF from PF images?

### DISCUSSION THREAD

**Discussion by Gbowker** [mtex-toolbox/mtex#1412] [Feature Requests]
Hello, MTEX is amazing, thank you very much for this super helpful software! I don't know what I would do without it!

Are there any methods in MTEX currently (or anywhere else for that matter) that is able to create an ODF object from multiple pole figure images? Similar to [datatheif III](https://datathief.org/) but instead of defining axes limits and datapoints defining crystal symmetry, lab axes, crystal planes/directions? This would be massivley useful to a lot of researchers I'm sure!
My motivation for this is for replicating initial textures from literature for use in a model. This will allow validation agianst flow-stress curves where they are given beside initial textures.

If you are not familiar with [datatheif III](https://datathief.org/), it is a program written in javascript to sample datapoints from an image of a graph:
![datatheif_screenshot](https://user-images.githubusercontent.com/61540494/165530738-2ee8faa3-92d4-49c9-8b34-839961fc755b.png)
[https://datathief.org/](https://datathief.org/)

What if this could be done for pole figures? For example input some pole figure images, define crystal symmetry, axes, etc. at the top, place markers at the axes and points of high texture strength and generate an ODF object?
<img width="684" alt="image" src="https://user-images.githubusercontent.com/61540494/165536996-aca12b8d-189b-4cd4-bf00-65051c73a467.png">

Just thought this was an idea worth sharing, thanks
Guy

**Comment by Begley-MSE** [comment]
Hi Guy,

@tori-miller and I just finished a tool that does almost exactly what you are looking for! We ran into the same kind of problem as you described, needing to validate simulation results against published textures, and ended up developing the tool ourselves because we couldn't find anything that could do it.

The GitHub repository is https://github.com/MONSTERgroup/DRAGON, if you want to check it out. We'd love feedback on it!

**Comment by ralfHielscher** [comment]
Hi Ben,
I was not aware of your project. It looks really great - congratulations! Won't you mind if we link you at the MTEX homepage? We are currently reworking the homepage and want to put more focus on the various extensions. Maybe you want to make an documentation page that describe its usage.

Ralf.

**Reply by Begley-MSE** [reply]
We're happy to have it on the MTEX homepage! I'm working on better documentation than just our GitHub readme, and a couple of video tutorials for it as well, but those are still in progress.
